src/eos_gates.js

```diff
@@ -143,6 +237,115 @@ const Application = new Lang.Class({
     },
 });
 
+function spawnProcess(argv=[]) {
+    return Gio.Subprocess.new(argv, Gio.SubprocessFlags.NONE);
+}
+
+function getAppStoreAppId(remote, appId) {
+    let installation = Flatpak.Installation.new_system(null);
+    let flatpakRemote = null;
+
+    try {
+        flatpakRemote = installation.get_remote_by_name(remote, null);
+    } catch (e) {
+        logError(e, 'Could not find flatpak remote %s: %s'.format(remote));
+    }
+
+    // Get the default branch now to construct the full unique ID GNOME Software expects.
+    if (flatpakRemote) {
+        let defaultBranch = flatpakRemote.get_default_branch()
+        if (defaultBranch)
+            return 'system/flatpak/%s/desktop/%s.desktop/%s'.format(remote,
+                                                                    appId,
+                                                                    defaultBranch);
+    }
+
+    return appId;
+}
+
+// Happens when we launch gnome-software for a known flatpak replacement
+// of an unsupported binary.
+const EVENT_LAUNCHED_INSTALLER_FOR_FLATPAK = 'e98bf6d9-8511-44f9-a1bd-a1d0518934b9';
+
+// Happens when we launch a known flatpak replacement of an unsupported binary
+// directly.
+const EVENT_LAUNCHED_EXISTING_FLATPAK = '192f39dd-79b3-4497-99fa-9d8aea28760c';
+
+// Happens when we launch gnome-software to install an 'equivalent' app for
+// an unsupported binary.
+const EVENT_LAUNCHED_EQUIVALENT_INSTALLER_FOR_FLATPAK = '7de69d43-5f6b-4bef-b5f3-a21295b79185';
+
+// Happens when we launch an 'equivalent' app for an unsupported binary directly. 
+const EVENT_LAUNCHED_EQUIVALENT_EXISTING_FLATPAK = '00d7bc1e-ec93-4c53-ae78-a6b40450be4a';
+
+function installAppFromStore(replacement, originalPayload) {
+    let appStoreId = getAppStoreAppId(replacement.flatpakInfo.remote,
+                                      replacement.flatpakInfo.id);
+    recordMetrics(replacement.replacementInfo ?
+                  EVENT_LAUNCHED_EQUIVALENT_INSTALLER_FOR_FLATPAK :
+                  EVENT_LAUNCHED_INSTALLER_FOR_FLATPAK,
+                  new GLib.Variant('(sas)', [replacement.flatpakInfo.id, originalPayload]));
+    spawnProcess(['gnome-software', '--details=%s'.format(appStoreId)]);
```

Comment:
Yep. I've implemented that, though it is a slightly more verbose way of doing what is essentially the same thing. We can keep the latter approach if it is desired though.